apps/events/models.py

```diff
@@ -793,8 +754,30 @@ def get_payment(self):
     def __str__(self):
         return self.event.title
 
+    def check_for_waitlist_changes(self):
```

Comment:
I'm a bit conflicted about this method.

I understand what it tries to do and I like it, but I think it should require one (or two (or one and one optional)) event(s) as arguments, so that it can be used outside of the `save`-method, which it now is tightly coupled to.

Like `def check_for_waitlist_changes(self, old_event)`, or even as a classmethod with `cls, updated_event, old_event` if that's your fancy.

Also, when I'm here: I don't like a function called `check_something` that actually does an action; I'd expect it to return a value I can use myself to do an action if the check is truthy (or falsy, for that matter), not send emails to some people. ;-)

Reply:
I get what you mean, but it's not really coupled to the save method. 
It would however not make any sense to use this without calling .save after though, but I can't think of any other ways to do this check without modifying the fields. 
Using a classmethod seems silly to me as you then have to manually call save on one of the objects after. 

I agree on the method name though. Any suggestions?

Reply:
True, but it could be used to inform people what is going to happen, e.g. if someone increases the capacity of the event, it could inform the user that "This action is going to give 5 attendees currently on waitlist space on this event. Confirm this?".

Classmethod was a stretch, disregard that.

What I meant was that `save()` could call this method, and if it's truthy it can do `notify_waiting_list()`:

```
def save(…):
    if self.check_for_waitlist_changes():
        self.notify_waiting_list()
    ...
    super().save()
```

was more along the lines of what I thought.

Reply:
Yeah, that makes sense. I could rewrite it to return the number of attendees bumped from the waitlist and call notify_waiting_list with that number if > 0.

Reply:
:+1: